lib/src/mobile_scanner.dart

```diff
@@ -270,33 +280,37 @@ class _MobileScannerState extends State<MobileScanner>
       );
     }
     if (controller.autoStart) {
-      controller.start();
+      await controller.start();
     }
-    super.initState();
   }
 
-  @override
-  void dispose() {
-    super.dispose();
-
-    if (_subscription != null) {
-      _subscription!.cancel();
-      _subscription = null;
-    }
+  Future<void> disposeMobileScanner() async {
+    await _subscription?.cancel();
+    WidgetsBinding.instance.removeObserver(this);
 
     if (controller.autoStart) {
-      controller.stop();
+      await controller.stop();
     }
-    // When this widget is unmounted, reset the scan window.
-    unawaited(controller.updateScanWindow(null));
 
     // Dispose default controller if not provided by user
     if (widget.controller == null) {
-      controller.dispose();
-      WidgetsBinding.instance.removeObserver(this);
+      await controller.dispose();
     }
   }
 
+  @override
+  void initState() {
+    super.initState();
+    controller = widget.controller ?? MobileScannerController();
+    unawaited(initMobileScanner());
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    disposeMobileScanner();
```

Comment:
Should we use unawaited here? (like in initState for the setup?)

```suggestion
    unawaited(disposeMobileScanner());
```